lld/ELF/LinkerScript.cpp

```diff
@@ -1196,9 +1202,17 @@ void LinkerScript::adjustOutputSections() {
     if (sec->sectionIndex != UINT32_MAX)
       maybePropagatePhdrs(*sec, defPhdrs);
 
+    // Discard .relro_padding if we have not seen one non-NOBITS RELRO section.
+    // Note: when .tbss is the only RELRO section, it's difficult to pick a
```

Comment:
I can see that a .tbss only (which would have no impact on the PT_LOAD) would be best excluded. It looks like the code would also exclude .bss.rel.ro only. Although I'm not entirely sure if any tool actually generates those.